lib/Sema/CSApply.cpp

```diff
@@ -5454,6 +5459,59 @@ static bool hasCurriedSelf(ConstraintSystem &cs, ConcreteDeclRef callee,
   return false;
 }
 
+static void checkNonEphemeralArgumentConversion(
```

Comment:
I think it would make most sense to make it a`TypeMatchFlag`. Maybe some constraints should just carry type match options which could be deduced by `matchCallArguments`? Or otherwise it has to be a "constraint restriction"...